### DISCUSSION TITLE
How to load robot system from the library using python

### DISCUSSION THREAD

**Discussion by Kenta042** [visose/Robots#286] [Q&A] [answered]
Hi I have a question regarding the visose. I was wondering if anyone knew there is a way to access the robot systems library similar to the LoadRobotSystem component but through ghpython. Any inputs is greatly appreciated!

**Comment by visose** [comment]
Hi @Kenta042, "visose" is the Github user name, "Robots" is the project name.

There's a Python example in the samples folder, not sure if you had a look at it: https://github.com/visose/Robots/blob/master/samples/Robots.Samples.Grasshopper/Python%20script%20example.gh

**Reply by Kenta042** [reply]
Sorry for the confusion for the name. Yes I did look into the file but it only contains the part of creating the toolpath using python not the part of actually loading the robot instances using python. Is there a way to do this part also through python?

![image](https://github.com/user-attachments/assets/5b279e6d-9fc8-4de0-a31d-858f1be862d7)

**Reply by visose** [reply]
Yes, you can use `robotSystem = FileIO.LoadRobotSystem(name, Plane.WorldXY, true, null)` where:
-  `FileIO` is a static class inside the `Robots` namespace.
-  `name` is the RobotCell (the name defined in the XML of the robot library).
-  `true`: if you want to load meshes, switch to false for no meshes.
- `null`: will use the default post-processor for this robot manufacturer, you can optionally pass a custom one.


Can I ask why do you want to load it this way?

**Comment by visose** [comment, marked as answer]
Use `FileIO.LoadRobotSystem(name, Plane.WorldXY, true, null)` from GhPython/C# after referencing/importing the Robots assembly/namespace. `name` is the `RobotCell` name from the XML, `true` loads meshes, and `null` uses the default postprocessor for that manufacturer. Closing this as answered.
